portal/server/helpers/constructPayload.js

```diff
@@ -5,16 +5,24 @@
  * @param {Object} body Request body (req.body)
  * @param {Array} properties Interested properties to be added into payload
  * @param {Boolean} csrf Include CSRF token, defaulted to `true`
+ * @param {Boolean} propertyEmpty List of property names that can't have empty value as they will have object value later, defaulted to True
  * @returns {Object} Payload
  */
-const constructPayload = (body, properties, csrf = true) => {
+const constructPayload = (body, properties, propertyEmpty, csrf = true) => {
   let payload = {};
-
+  const bodyCopy = { ...body };
   // Return empty payload upon void mandatory arguments
   if (!body || !properties) {
     return payload;
   }
 
+  // Currency will be an object, don't save as empty string, because MongoDB can't change the type.
```

Comment:
Not sure if we should mention `currency` in comment if we go with generic change. Remove empty properties if flag is set to true. This is because some fields might change type, for example currency field, and MongoDB doesn't allow type change.

Reply:
Since the function is generic in nature please re-write the comment.

Reply:
I checked and because of new minimalistic code comment was removed.